### PR TITLE
Fix waterfall X value for quantitative X axis

### PR DESCRIPTION
**Pull request by ariya**
Steps to try:

1. Ask a question, Native query
2. Choose Sample Dataset
3. Type the following and then Run query (Ctrl+Enter)

```sql
select 1 as x, 10 as y
union
select 2 as x, -2 as y
```

4. Visualization, Waterfall
5. Settings, Display, Show values on data points

**Expected**: The waterfall chart should look like the following screenshot:

image

### CODE REVIEW

File: frontend/src/metabase/visualizations/lib/renderer_utils.js [hunk 1]
```diff
@@ -322,11 +322,15 @@ export function getXInterval({ settings, series }, xValues, warn) {
...
+  const xValues = getXValues({ settings, series });
+  const xInterval = getXInterval({ settings, series }, xValues);
+
   if (isTimeseries(settings)) {
-    const xValues = getXValues({ settings, series });
-    const { count, interval } = getXInterval({ settings, series }, xValues);
+    const { count, interval } = xInterval;
     const lastXValue = xValues[xValues.length - 1];
     return lastXValue.clone().add(count, interval);
+  } else if (typeof xInterval === "number") {
```
Comment (ariya) [inline frontend/src/metabase/visualizations/lib/renderer_utils.js:332]: This check feels whacky. @paulrosenzweig is there a better alternative?

Comment (paulrosenzweig) [reply frontend/src/metabase/visualizations/lib/renderer_utils.js:332]: Yup! I think `isQuantitative` is what you want.

Comment (paulrosenzweig) [reply frontend/src/metabase/visualizations/lib/renderer_utils.js:332]: I see `isQuantitative(settings) || isHistogram(settings)` is used in `getXInterval`, so it's worth checking if there's an edge case for histograms. I'm not sure what a non-quantitative histogram would be...

**Review by paulrosenzweig — APPROVED** [review]
Besides the `isQuantitative` check, this looks good!

I think the title has a typo though:
Fix waterfall X value for ~non~-quantitative X axis

File: frontend/src/metabase/visualizations/lib/renderer_utils.js (hunk 1 above)
Comment (ariya) [reply frontend/src/metabase/visualizations/lib/renderer_utils.js:332]: The histogram looks funny but it definitely works. Meanwhile, I'll adopt your proposed checks.

image

**Comment by ariya** [comment]
> I think the title has a typo though:

Fixed! (thought I had to keep the branch, to avoid redoing the PR)

**Comment by salsakran** [comment]
the fonts on the values within the bars look pretty off. @kdoh @mazameli is that as designed or do we need to tweak it?

**Comment by mazameli** [comment]
@salsakran Need to tweak it. I noted this as a bullet in [my comment on the implementation issue](https://github.com/metabase/metabase/issues/13661#issuecomment-740267819) yesterday, but I've just added an accompanying screenshot for clarity. Here's the screenshot:

image

**Pull request merged**